app/guard/guard.go

```diff
@@ -0,0 +1,256 @@
+// Guard is a service that provides additional password protection by creating a series of guard servers that allow a
+// Client to further secure their passwords from direct database compromises. The service is hash based and the passwords
+// never leave the main database, making the guard servers very lightweight. The guard server's are used in both setting and
+// authenticating passwords.
+
+package main
+
+import (
+	"os"
+
+	"errors"
+
+	"crypto/aes"
+	"crypto/cipher"
+	"crypto/rand"
+
+	"github.com/dedis/cothority/app/lib/config"
+	"github.com/dedis/cothority/log"
+	"github.com/dedis/cothority/sda"
+	"github.com/dedis/cothority/services/guard"
+	"github.com/dedis/crypto/abstract"
+
+	"io/ioutil"
+
+	s "github.com/SSSaaS/sssa-golang"
+	"gopkg.in/codegangsta/cli.v1"
+
+	"bytes"
+
+	"github.com/dedis/cothority/network"
+)
+
+// User is a representation of the Users data in the code, and is used to store all relevant information
+type User struct {
+	// Name or UserID
+	Name []byte
+	// Salt used for the password-hash
+	Salt []byte
+	// Xored keys with hash
+	Keys [][]byte
+	// Data AEAD-encrypted with key
+	Data []byte
+}
+
+// Database is a structure that stores Cothority(the list of guard servers), and a list of all users within the database
+type Database struct {
+	Cothority *sda.Roster
+	Users     []User
+}
+
+var db *Database
+
+var mastersalt []byte
+
+func main() {
+	network.RegisterMessageType(&Database{})
+	salty := make([]byte, 12)
+	rand.Read(salty)
+	mastersalt = salty
+	app := cli.NewApp()
+	app.Name = "Guard"
+	app.Usage = "Get and print status of all servers of a file."
+
+	app.Flags = []cli.Flag{
+		cli.StringFlag{
+			Name:  "group, gd",
+			Value: "group.toml",
+			Usage: "Cothority group definition in `FILE.toml`",
+		},
+		cli.IntFlag{
+			Name:  "debug, d",
+			Value: 0,
+			Usage: "debug-level: `integer`: 1 for terse, 5 for maximal",
+		},
+	}
+	app.Commands = []cli.Command{
+		{
+			Name:    "setpass",
+			Aliases: []string{"s"},
+			Usage:   "Setup the configuration for the server (interactive)",
+			Action:  setpass,
+		},
+		{
+			Name:      "setup",
+			Aliases:   []string{"su"},
+			Usage:     "Saves the cothority group-toml to the configuration",
+			ArgsUsage: "Give group definition",
+			Action:    setup,
+		},
+		{
+			Name:    "recover",
+			Aliases: []string{"r"},
+			Usage:   "Gets the password back from the guards",
+			Action:  get,
+		},
+	}
+	app.Before = func(c *cli.Context) error {
+		b, err := ioutil.ReadFile("config.bin")
+		if os.IsNotExist(err) {
+			return nil
+		}
+		log.ErrFatal(err, "The config.bin file threw an error")
+		_, msg, err := network.UnmarshalRegistered(b)
+		log.ErrFatal(err, "UnmarshalRegistered messeed up")
+		var ok bool
+		db, ok = msg.(*Database)
+		if !ok {
+			log.Fatal("We are not OK")
+		}
+		return nil
+	}
+	app.Run(os.Args)
+}
+
+func xor(x []byte, y []byte) []byte {
```

Comment:
helper functions to the bottom, most important functions to the top. So `xor` should be at the bottom.